### DISCUSSION TITLE
Having both an old version and new version of the same dependency.

### DISCUSSION THREAD

**Discussion by nielsbasjes** [renovatebot/renovate#10340] [Request Help] [answered]
In a public project of mine ( https://github.com/nielsbasjes/yauaa ) I have integration tests that verify if my library works in combination with Kryo 4.x and 5.x.
See:
- https://github.com/nielsbasjes/yauaa/blob/master/pom.xml#L96
- https://github.com/nielsbasjes/yauaa/tree/master/analyzer-it/serialization

So I have multiple integration tests using different versions of the same dependency in different modules of the same multi-module project.
I want both to be updated automatically.
Yet the `kryo4.version` must remain a 4.x version and the `kryo5.version` must remain a 5.x version.

For me it is also fine to never update the kryo4.

I have been unable to figure out how to configure this correctly.
I have tried this but somehow this did not do what I thought it would do.
```json
{
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "matchDepTypes": [
        "kryo4.version",
        "elasticsearch.log4j2.version"
      ],
      "rangeStrategy": "pin"
    }
  ]
}
```

Please advise.

**Comment by rarkins** [comment, marked as answer]
Can you locate the `packageFiles with updates` log message and narrow in on the `kryo4.version` entry? I want to confirm its `depName` first.

I'll let you know the exact config later, but you can choose between:
- Simply disabling all updates for that version, or
- Disabling major updates but allowing any minor or patch if they occur

**Reply by nielsbasjes** [reply]
In this case I would like the "Disabling major updates but allowing any minor or patch if they occur".

I think this is the part you want:
```
          {
            "datasource": "maven",
            "depName": "com.esotericsoftware:kryo",
            "currentValue": "4.0.2",
            "fileReplacePosition": 3640,
            "registryUrls": [
              "https://repo.maven.apache.org/maven2"
            ],
            "depType": "test",
            "groupName": "kryo4.version",
            "editFile": "pom.xml",
            "depIndex": 8,
            "updates": [
              {
                "bucket": "major",
                "newVersion": "5.1.1",
                "newValue": "5.1.1",
                "releaseTimestamp": "2021-05-02T09:17:26.000Z",
                "newMajor": 5,
                "newMinor": 1,
                "updateType": "major",
                "branchName": "renovate/major-all"
              }
            ],
            "warnings": [],
            "currentVersion": "4.0.2",
            "isSingleVersion": true,
            "fixedVersion": "4.0.2"
          },

```

**Reply by rarkins** [reply]
Something like this:

```json
{
  "packageRules": [{
    "description": "Disable major updates for kryo4",
    "matchPackageNames": ["com.esotericsoftware:kryo"],
    "matchCurrentVersion": "[4.0.0,5.0.0)", // please check if I've remembered maven range syntax correctly
    "matchUpdateTypes": ["major"],
    "enabled": false
  }]
}
```

**Reply by nielsbasjes** [reply]
Yes! I added this and the existing pull request for this change got autoclosed.
https://github.com/nielsbasjes/yauaa/blob/master/renovate.json

Thanks !

**Reply by nielsbasjes** [reply]
The code you provided works great.
What I see here is that you disable major updates for the mentioned package if the version is 4.x .
I was wondering if it is also possible not to pin this to the package name but to the variable name that is used to store it in?

**Reply by rarkins** [reply]
No, not at this point - we don't pass the variable name through as a defined field or have a way of matching based on it

**Reply by nielsbasjes** [reply]
Ok, I noticed the `"groupName": "kryo4.version",` and thought that may be a route to do something like that.

Thanks for the great tool!

**Reply by rarkins** [reply]
Yeah, that's actually a little bit of a hack which I'd like to replace with an explicit `variableName` field in future, after which we could add a `matchVariableNames` field.
